frontend/charm/tox.ini

```diff
@@ -66,7 +66,7 @@ commands =
 description = Run integration tests
 deps =
     pytest
-    juju
+    juju<=3.2
```

Comment:
This is intentionally `<= 3.2` as opposed to for example `>= 2.9` as is set in the charm metadata: the integration test environment needs to be (for reasons of various version incompatibilities) 3.x.